### ISSUE TITLE
Download datasets

### ISSUE THREAD

**Issue by MohammedBadaruddin** [MohammedBadaruddin/rman-autonomous#2]
**KRUTIKA:** 

### **Download and prepare the datasets required for perception and object detection.**

Datasets:
- KITTI for object detection.
- BDD100K for YOLOv8 training.
- nuScenes mini for multi-sensor fusion demonstration.

Tasks:
- Store datasets in /data/raw/.
- Create train/validation/test splits using:
  70% training
  20% validation
  10% testing

Deliverable:
/data/raw/ containing the downloaded and organized datasets.

### **Set up MLflow experiment tracking:**

Set up MLflow for tracking machine learning experiments.

Tasks:
- Install MLflow.
- Start the MLflow server.
- Log every training experiment.
- Record:
  - Model name
  - Number of epochs
  - mAP
  - Loss

Suggested command:
mlflow server --host 0.0.0.0 --port 5000

Deliverable:
MLflow dashboard accessible at localhost:5000 with experiments being logged.

### **### Download & prepare Fine-tune YOLOv8 on BDD100K datasets:**
Fine-tune the YOLOv8 model using the BDD100K dataset.

Tasks:
- Use YOLOv8m pretrained model.
- Train using the prepared BDD100K dataset.
- Train for 50 epochs.
- Use image size 640.
- Track training experiments using MLflow.

Target classes:
- Vehicle
- Pedestrian
- Traffic sign
- Obstacle

Target:
mAP@0.5 >= 85%

Deliverable:
Trained YOLOv8 model (.pt file) with mAP scores logged in MLflow.

### **Implement Kalman Filter sensor fusion:**

Implement Kalman Filter-based multi-sensor fusion.

Tasks:
- Use FilterPy KalmanFilter.
- Fuse camera object information with simulated LiDAR/depth information from CARLA.
- Use the state:
  [x, y, vx, vy]
- Compare single-sensor results with fused sensor results.
- Measure position error reduction using RMSE.

Deliverable:
fusion.py and an RMSE comparison between single-sensor and fused-sensor results.
